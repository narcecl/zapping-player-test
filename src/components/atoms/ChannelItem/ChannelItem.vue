<script setup lang="ts">
import { computed } from 'vue';
import { useLanguageStore } from '@/stores';
import ChannelLogo from '@/components/atoms/ChannelLogo/ChannelLogo.vue';
import type { Channel } from '@/interfaces/channel';

const { name, active } = defineProps<Channel>();
const { $t } = useLanguageStore();

const componentClasses = computed(() => ({
    channer__item: true,
    'channer__item--active': active,
}));
</script>

<template>
    <button :class="componentClasses" :aria-label="$t('accessibility_change_channel', name)">
        <p class="channer__item__number">
            {{ number }}
        </p>

        <ChannelLogo :name="name" :logo="logo_color" size="large" />

        <h3 class="channer__item__name text-primary">
            {{ name }}
        </h3>
    </button>
</template>

<style lang="scss" scoped>
.channer__item {
    width: 100%;
    align-items: center;
    background: rgba(255, 255, 255, 0.05);
    border-radius: 12px;
    border: 2px solid transparent;
    display: flex;
    gap: 16px;
    padding: 24px 32px;
    transition: all 0.3s ease;
    opacity: 0.6;
    cursor: pointer;

    &__number,
    &__name {
        color: white;
    }

    &__number {
        min-width: 32px;
        text-align: center;
    }

    &__name {
        text-align: left;
        max-width: 110px;
    }

    &:hover {
        opacity: 1;
    }

    &--active {
        background: rgba(255, 255, 255, 0.1);
        border-color: white;
        opacity: 1;
    }
}
</style>
