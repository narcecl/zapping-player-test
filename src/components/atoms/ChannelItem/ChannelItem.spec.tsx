import { describe, test, expect, vi } from 'vitest';
import { createTestingPinia } from '@pinia/testing';
import { mount } from '@vue/test-utils';
import ChannelItem from './ChannelItem.vue';
import type { Channel } from '@/interfaces/channel';

const CHANNEL_DEMO: Channel = {
    active: true,
    currentShow: {},
    logo_color: 'https://davinci.zappingtv.com/gato/media/256/canales/color/sony.jpg',
    logo_white: 'https://davinci.zappingtv.com/gato/media/256/canales/white/sony.png',
    name: 'Sony Channel',
    number: 103,
    video: '',
};

describe('Badge', () => {
    test('Default render (button)', () => {
        const wrapper = mount(ChannelItem, {
            global: {
                plugins: [
                    createTestingPinia({
                        createSpy: vi.fn,
                    }),
                ],
            },
            props: {
                ...CHANNEL_DEMO,
            },
        });
        expect(wrapper.html()).toMatchSnapshot();

        const title = wrapper.get('h3');
        expect(title).toBeDefined();
        expect(title.text()).toEqual(CHANNEL_DEMO.name);

        const channelNumner = wrapper.get('.channer__item__number');
        expect(channelNumner).toBeDefined();
        expect(channelNumner.text()).toBe(`${CHANNEL_DEMO.number}`);

        const channelLogo = wrapper.get('img');
        expect(channelLogo).toBeDefined();
        expect(channelLogo.attributes('src')).toBe(CHANNEL_DEMO.logo_color);

        expect([...wrapper.element.classList]).includes('channer__item');
    });
});
