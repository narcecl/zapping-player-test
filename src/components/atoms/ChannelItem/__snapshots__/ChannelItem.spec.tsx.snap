// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Badge > Default render (button) 1`] = `
"<button data-v-90b62ae5="" class="channer__item channer__item--active">
  <p data-v-90b62ae5="" class="channer__item__number">103</p>
  <picture data-v-922226ac="" data-v-90b62ae5="" class="channer__item__logo channer__item__logo--large"><img data-v-922226ac="" src="https://davinci.zappingtv.com/gato/media/256/canales/color/sony.jpg"></picture>
  <h3 data-v-90b62ae5="" class="channer__item__name text-primary">Sony Channel</h3>
</button>"
`;
